tracer/tracer.go

```diff
@@ -82,28 +84,37 @@ func New(config Config) (*Tracer, error) {
 	if len(t.WorkflowFuncs) != 1 {
 		return nil, fmt.Errorf("single workflow function required")
 	}
-	// TODO(cretz): Support struct-based workflows
 	lastDot := strings.LastIndex(config.WorkflowFuncs[0], ".")
 	if lastDot == -1 {
 		return nil, fmt.Errorf("workflow function missing dot")
 	}
 	t.fnPkg, t.fn = config.WorkflowFuncs[0][:lastDot], config.WorkflowFuncs[0][lastDot+1:]
+	// check for struct-based workflow function
+	base := path.Base(config.WorkflowFuncs[0])
+	if strings.Count(base, ".") > 2 {
```

Comment:
I don't think this will work for `github.com/myorg/myrepo/my.dir/mypkg.Func` right? The code will have to change to probably take the first dot after the last slash to separate package from function. And then the function itself can be parsed.

Reply:
Base is a controversial name. It returns counterintuitively the last path element. https://pkg.go.dev/path#Base

It will work for your example as well.